internal/internal_event_handlers.go

```diff
@@ -728,8 +733,24 @@ func (wc *workflowEnvironmentImpl) GetVersion(changeID string, minSupported, max
 		// GetVersion for changeID is called first time (non-replay mode), generate a marker command for it.
 		// Also upsert search attributes to enable ability to search by changeVersion.
 		version = maxSupported
-		wc.commandsHelper.recordVersionMarker(changeID, version, wc.GetDataConverter())
-		_ = wc.UpsertSearchAttributes(createSearchAttributesForChangeVersion(changeID, version, wc.changeVersions))
+		changeVersionSA := createSearchAttributesForChangeVersion(changeID, version, wc.changeVersions)
+		attr, err := validateAndSerializeSearchAttributes(changeVersionSA)
```

Comment:
I know we never cared about serialization failures before before (and it should never happen), but IMO we should either log it or truly ignore it and don't let it affect code path

Reply:
>  don't let it affect code path

not sure what you mean by this. If we can't serialize the SA we can't check the size or upsert so it will effect the code path. I am fine logging, panicing or skipping the upsert but still updating the marker personally.

Reply:
Logging works for me. I just meant ignore because you know it will never fail, but I too prefer logging.